harness/determined/pytorch/lightning/_adapter.py

```diff
@@ -299,12 +300,11 @@ def lightning_scheduler_dict_to_det(lrs: dict) -> _LRScheduler:
                     "- Tuple of dictionaries as described, with an optional ‘frequency’ key.\n"
                 )
 
-            check.check_isinstance(
-                lrs["scheduler"].optimizer,
-                Optimizer,
-                "A returned LRScheduler from `configure_optimizers` is "
-                "missing the optimizer attribute.",
-            )
+            if not isinstance(lrs["scheduler"].optimizer, Optimizer):
+                raise TypeError(
+                    "A returned LRScheduler from `configure_optimizers` is "
+                    "missing the optimizer attribute.",
+                )
```

Comment:
not your fault, but the message and the check are not at all aligned.

If `lrs["scheduler"]` doesn't have a `.optimizer` attribute, an error is raised before the `isinstance()` can even evaluate.  So if `isinstance()` does evaluate to false the message is guaranteed to be wrong.

Reply:
Looking at the surrounding code, the check doesn't make sense to me, regardless of the message.
Anyway, I've changed the message to better reflect the check.